letsencrypt/cli.py

```diff
@@ -1051,6 +1030,51 @@ def _plugins_parsing(helpful, plugins):
 
     helpful.add_plugin_args(plugins)
 
+    # These would normally be a flag within the webroot plugin, but because
+    # they are parsed in conjunction with --domains, they live here for
+    # legibiility. helpful.add_plugin_ags must be called first to add the
+    # "webroot" topic
+    helpful.add("webroot", "-w", "--webroot-path", action=WebrootPathProcessor,
+                help="public_html / webroot path")
+    parse_dict = lambda s: dict(json.loads(s))
+    helpful.add("webroot", "--webroot-map", default={}, type=parse_dict,
```

Comment:
I feel that having the user write JSON on the command line is awkward, especially when they still have to specify all of their domains again with `-d/--domains`. To me, this feels like an unnecessarily exposed implementation detail that we should hide from the client.